File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        pre = [1] * (n + 1)
        suf = [1] * (n + 1)
        for i in range(n):
            pre[i + 1] = pre[i] * nums[i]
        
        for i in range(n - 1, -1, -1):
            suf[i] = suf[i + 1] * nums[i]
        
        res = []
        for i in range(n):
            res.append(pre[i] * suf[i+1])
        return res

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> freq;
        for (int num: nums) {
            freq[num]++;
        }
        priority_queue<pair<int, int>> pq;
        for (auto [key, val]: freq) {
            pq.push({val, key});
        }
        vector<int> res;
        for (int i = 0; i < k; i++) {
            auto [c, num] = pq.top();
            res.push_back(num);
            pq.pop();
        }
        return res;
    }
};
